tools/rpkg/tests/testthat/test_relational.R

```diff
@@ -123,3 +123,66 @@ test_that("we can get the relation object back from an altrep df", {
   rel2 <- rel_from_altrep_df(df)
   expect_true(TRUE)
 })
+
+test_that("Union all does not immediately materialize", {
+    test_df_a <- rel_from_df(con, data.frame(a=c('1', '2'), b=c('3', '4')))
+    test_df_b <- rel_from_df(con, data.frame(a=c('5', '6'), b=c('7', '8')))
+    rel <- rel_union_all(test_df_a, test_df_b)
+    rel_df <- rel_to_altrep(rel)
+    expect_false(df_is_materialized(rel_df))
+    dim(rel_df)
+    expect_true(df_is_materialized(rel_df))
+})
+
+test_that("Union all has the correct values", {
+    test_df_a <- rel_from_df(con, data.frame(a=c('1', '2'), b=c('3', '4')))
+    test_df_b <- rel_from_df(con, data.frame(a=c('5', '6'), b=c('7', '8')))
+    rel <- rel_union_all(test_df_a, test_df_b)
+    rel_df <- rel_to_altrep(rel)
+    expect_false(df_is_materialized(rel_df))
+    dim(rel_df)
+    expect_true(df_is_materialized(rel_df))
+    expected_result = data.frame(a=c('1', '2', '5', '6'), b=c('3', '4', '7', '8'))
```

Comment:
The assignment operator here is inconsistent.

```suggestion
    expected_result <- data.frame(a=c('1', '2', '5', '6'), b=c('3', '4', '7', '8'))
```

You can style the R code with `styler::style_pkg()` too.